pkg/proxy/handlers/spacelister_get_test.go

```diff
@@ -726,3 +742,234 @@ func TestGetUserWorkspace(t *testing.T) {
 		})
 	}
 }
+
+func TestSpaceListerGetPublicViewerEnabled(t *testing.T) {
+
+	fakeSignupService := fake.NewSignupService(
+		newSignup("batman", "batman.space", true),
+		newSignup("robin", "robin.space", true),
+	)
+
+	fakeClient := fake.InitClient(t,
+		// space
+		fake.NewSpace("robin", "member-1", "robin"),
+		fake.NewSpace("batman", "member-1", "batman"),
+
+		// spacebindings
+		fake.NewSpaceBinding("robin-1", "robin", "robin", "admin"),
+		fake.NewSpaceBinding("batman-1", "batman", "batman", "admin"),
+
+		fake.NewSpaceBinding("community-robin", toolchainv1alpha1.KubesawAuthenticatedUsername, "robin", "viewer"),
+	)
+
+	robinWS := workspaceFor(t, fakeClient, "robin", "admin", true)
+	batmanWS := workspaceFor(t, fakeClient, "batman", "admin", true)
+	publicRobinWS := func() *toolchainv1alpha1.Workspace {
+		batmansRobinWS := robinWS.DeepCopy()
+		batmansRobinWS.Status.Type = ""
+		batmansRobinWS.Status.Role = "viewer"
+		return batmansRobinWS
+	}()
+	tests := map[string]struct {
+		username          string
+		expectedErr       string
+		workspaceRequest  string
+		expectedWorkspace *toolchainv1alpha1.Workspace
+	}{
+		"robin can get robin workspace": {
+			username:          "robin.space",
+			workspaceRequest:  "robin",
+			expectedWorkspace: &robinWS,
+		},
+		"batman can get batman workspace": {
+			username:          "batman.space",
+			workspaceRequest:  "batman",
+			expectedWorkspace: &batmanWS,
+		},
+		"batman can get robin workspace as public-viewer": {
+			username:          "batman.space",
+			workspaceRequest:  "robin",
+			expectedWorkspace: publicRobinWS,
+		},
+		"robin can not get batman workspace": {
+			username:          "robin.space",
+			workspaceRequest:  "batman",
+			expectedWorkspace: nil,
+			expectedErr:       "",
+		},
+		"gordon can get batman workspace": {
```

Comment:
Hm... How can gordon get the batman workspace? I don't see the SpaceBinding declared for gordon and the batman workspace... What am I missing?

Reply:
that's a typo in the title, great catch! It should be `gordon can get robin workspace`. robin workspace is actually shared with the community. Fixed in 873612414d8de8307a1a4df96e2749a96673e7f8